types/mdast/index.d.ts

```diff
@@ -1,27 +1,103 @@
 // Type definitions for Mdast 3.0
 // Project: https://github.com/syntax-tree/mdast, https://github.com/wooorm/mdast
 // Definitions by: Jun Lu <https://github.com/lujun2>
+//                 Thomas "zemnmez" Shadwell <https://github.com/zemnmez>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 import { Parent as UnistParent, Literal as UnistLiteral, Node } from 'unist';
 
+// typing notes:
+//
+// 1. MDAST does not explicitly specify a difference between a field which is undefined, a field
+//    which is null, and a field which is not present. However, the examples *do* contain `null`,
+//    so for safety's sake I am nulling every potentially not present field, and all fields specifying
+//    undefined also potentially specify null.
+//
+// 2. the 'root' node allows any kind of Content as a child, as long as that is the consistent type of the
+//    child i.e. all children can be 'TopLevel' block content like paragraphs OR table rows, but not both.
+//
+//    To better express this in typescript, Root is a union of RootT interfaces with the selected types.
+//    This should have the benefit of making it easier to discriminate between different root types.
+//
+// 3. In MDAST's definition, List is a type with rather complex field relationships -- some fields are only
+//    potentially present if other fields have some value. To express this better, I, similarly to 'root'
+//    express List in terms of a type union of OrderedList and UnorderedList.
+//
+// 4. It's definitely arguable that these typings should focus more on ease of use, rather than accurately reflecting
+//    the mdast spec documentation. I've picked somewhat of a middleground by making the most minor changes I can
+//    where I think it's useful.
+
+// explicit exports
+export {};
```

Comment:
We probably don't need this. Let's finish discussing in https://github.com/unifiedjs/unified-message-control/pull/7#discussion_r368266506 and come back to this pr later.